cpp/cmake_modules/DefineOptions.cmake

```diff
@@ -144,6 +144,8 @@ if("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_CURRENT_SOURCE_DIR}")
 
   define_option(ARROW_ORC "Build the Arrow ORC adapter" OFF)
 
+  define_option(ARROW_ORC_JNI "Build the Arrow ORC JNI lib" OFF)
```

Comment:
What do you think about having an `ARROW_WITH_JNI` option that will build the JNI bindings for any component that has them available?

Reply:
Different component may have different native lib dependency, I am not sure if a single flag will be sufficient. Let's say today I add orc JNI interface which require arrow_orc_jni, later parquet JNI is added which requires arrow_parquet_jni. How could a single flag deal with such a situation?

Reply:
Well, in such cases if we have

`if (ARROW_PARQUET AND ARROW_WITH_JNI)`

then the Parquet JNI bindings will be built. Similarly

`if (ARROW_ORC AND ARROW_WITH_JNI)`

It's conceivable that someone would want to build one set of JNI bindings but not another, so I'm not sure